libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -606,6 +611,59 @@ internal object MapboxRouteLineUtils {
         return itemsToReturn
     }
 
+    internal fun extractViolatedSectionsData(
+        route: NavigationRoute,
+        distancesProvider: (NavigationRoute) -> RouteLineGranularDistances?
+    ): List<ExtractedRouteRestrictionData> {
+        val itemsToReturn = mutableListOf<ExtractedRouteRestrictionData>()
+        val granularDistances = distancesProvider(route)
+        route.directionsRoute.legs()?.forEachIndexed { legIndex, leg ->
+            val legNotificationIndices = extractNotificationIndices(leg)
+            val legDistances = granularDistances?.legsDistances?.getOrNull(legIndex)
+            val routeDistance = granularDistances?.completeDistance
+            legNotificationIndices.forEach { (startIndex, endIndex) ->
+                val startRouteLineDistance = legDistances?.getOrNull(startIndex)
+                val endRouteLineDistance = legDistances?.getOrNull(endIndex)
+                if (startRouteLineDistance != null && endRouteLineDistance != null) {
+                    val startOffset = 1.0 - startRouteLineDistance.distanceRemaining / routeDistance!!
```

Comment:
If granularDistances above is nullable doesn't that also mean granularDistances?.completeDistance can be null? Here you're assuming it won't be.

Reply:
But then startRouteLineDistance would be null and we wouldn't have entered this branch.